log/log_test.go

```diff
@@ -0,0 +1,215 @@
+package log
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"github.com/rs/zerolog/log"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/rs/zerolog"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestEnvVarLogLevel(t *testing.T) {
+	defer os.Unsetenv("LOG_LEVEL")
+
+	type levelsToTestStruct struct {
+		stringLevel string
+		level       zerolog.Level
+	}
+
+	levelsToTest := []levelsToTestStruct{
+		{
+			stringLevel: "FATAL",
+			level:       zerolog.FatalLevel,
+		},
+		{
+			stringLevel: "fatal",
+			level:       zerolog.FatalLevel,
+		},
+		{
+			stringLevel: "FaTaL",
+			level:       zerolog.FatalLevel,
+		},
+		{
+			stringLevel: "ERROR",
+			level:       zerolog.ErrorLevel,
+		},
+		{
+			stringLevel: "WARN",
+			level:       zerolog.WarnLevel,
+		},
+		{
+			stringLevel: "INFO",
+			level:       zerolog.InfoLevel,
+		},
+		{
+			stringLevel: "DEBUG",
+			level:       zerolog.DebugLevel,
+		},
+		{
+			stringLevel: "TRACE",
+			level:       zerolog.TraceLevel,
+		},
+		{
+			stringLevel: "0",
+			level:       zerolog.FatalLevel,
+		},
+		{
+			stringLevel: "1",
+			level:       zerolog.ErrorLevel,
+		},
+		{
+			stringLevel: "2",
+			level:       zerolog.WarnLevel,
+		},
+		{
+			stringLevel: "3",
+			level:       zerolog.InfoLevel,
+		},
+		{
+			stringLevel: "4",
+			level:       zerolog.DebugLevel,
+		},
+		{
+			stringLevel: "5",
+			level:       zerolog.TraceLevel,
+		},
+		{
+			stringLevel: "invalid",
+			level:       zerolog.InfoLevel,
+		},
+	}
+
+	for _, levelToTest := range levelsToTest {
+		os.Setenv("LOG_LEVEL", levelToTest.stringLevel)
+		assert.Equal(t, levelToTest.level, resolveLogLevelFromEnv(), fmt.Sprintf("LOG_LEVEL=%v,levelToTest=%+v", os.Getenv("LOG_LEVEL"), levelToTest))
+	}
+}
+
+func TestLogRegression(t *testing.T) {
+
+	type loggedMessageAsJsonStruct struct {
+		Level   string
+		Time    string
+		Message string
+	}
+
+	type logOutputTestStruct struct {
+		envSettings        map[string]string
+		expectedLogMessage string
+	}
+
+	tests := []logOutputTestStruct{
+		{
+			envSettings: map[string]string{
+				"LOG_FORMAT": "text",
+			},
+			expectedLogMessage: "INF Kiali logging test 0: map[LOG_FORMAT:text]",
+		},
+		{
+			envSettings: map[string]string{
+				"LOG_FORMAT":            "text",
+				"LOG_LEVEL":             "debug",
+				"LOG_TIME_FIELD_FORMAT": time.RFC1123Z,
+			},
+			expectedLogMessage: "INF Kiali logging test 1: map[LOG_FORMAT:text LOG_LEVEL:debug LOG_TIME_FIELD_FORMAT:Mon, 02 Jan 2006 15:04:05 -0700]",
+		},
+		{
+			envSettings: map[string]string{
+				"LOG_FORMAT":            "json",
+				"LOG_LEVEL":             "debug",
+				"LOG_TIME_FIELD_FORMAT": time.RFC1123Z,
+			},
+			expectedLogMessage: "Kiali logging test 2: map[LOG_FORMAT:json LOG_LEVEL:debug LOG_TIME_FIELD_FORMAT:Mon, 02 Jan 2006 15:04:05 -0700]",
+		},
+	}
+
```

Comment:
Because the test is clearing the entire environment, it will affect all future unit tests that are run after this one executes. So far, all unit tests pass, but this could get ugly in the future with this bug being VERY hard to find ("where did all my env vars go??") :)

Here's a fix - using "defer", this unit test will put the environment back to the way it was before this test function executes so future unit tests don't lose their environment. 

```suggestion

	originalEnv := os.Environ()
	defer func() {
		for _, kvp := range originalEnv {
			kvpArray := strings.Split(kvp, "=")
			os.Setenv(kvpArray[0], kvpArray[1])
		}
		return
	}()
```